File: packages/domain/src/ports/IInvoiceSeeder.ts
import type { InvoiceStatus } from '@monolegal/shared';

export interface SeedClientInput {
  id: string;
  name: string;
  email: string;
}

export interface SeedInvoiceInput {
  clientId: string;
  invoiceNumber: string;
  concept: string;
  amount: number;
  dueDate: Date;
  status: InvoiceStatus;
}

export interface IInvoiceSeeder {
  resetAndSeed(clients: SeedClientInput[], invoices: SeedInvoiceInput[]): Promise<number>;
}

File: backend/dotnet/src/Monolegal.Domain/Ports/IInvoiceSeeder.cs
using Monolegal.Shared;

namespace Monolegal.Domain.Ports;

public record SeedClientInput(string Id, string Name, string Email);

public record SeedInvoiceInput(
    string ClientId,
    string InvoiceNumber,
    string Concept,
    decimal Amount,
    DateTime DueDate,
    InvoiceStatus Status);

public interface IInvoiceSeeder
{
    Task<int> ResetAndSeedAsync(IReadOnlyList<SeedClientInput> clients, IReadOnlyList<SeedInvoiceInput> invoices, CancellationToken cancellationToken = default);
}
